common/luaobject.c

```diff
@@ -304,8 +304,10 @@ luaA_object_emit_signal(lua_State *L, int oud,
             lua_remove(L, - nargs - nbfunc - 2 + i);
             luaA_dofunction(L, nargs + 1, 0);
         }
-    } else
-        warn("Trying to emit unknown signal '%s'", name);
+    } else {
+        luaA_warn(L, "Trying to emit unknown signal '%s'", name);
+        return;
```

Comment:
Well, what behavior really changes? When you do `c:emit_signal("foo")`, the "no such signal"-message is now only printed once instead of twice. I don't think this has any other effect than that. (But yeah, it's a behavior change)